contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpAPIConfig.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.store.http;
```

Comment:
The name of this package is `http`, which suggests it handles multiple kinds of HTTP requests, including REST, maybe ad-hoc formats, maybe other things.

In fact, this is a "generic REST" plugin. Would recommend that names reflect that to avoid confusion, especially if we end up adding other HTTP-based plugins in the future.

That is, the name should reflect the overall format (generic REST), not the transport (HTTP) which may be common to multiple plugins.

Reply:
I'm on the fence about this.  I'm calling it REST, but it really is HTTP -> JSON as you really could query anything that accepts HTTP and returns JSON.

Reply:
@paul-rogers 
I tend to take the pragmatic approach here.  The overarching goal was to enable Drill to get data that was stored remotely.  General observations was that most REST APIs return JSON, but clearly some return other data types.  

My thought is that in the future, we could add additional functionality such that this plugin could read XML, CSV and other data types that make sense. 

Also, the original name for this plugin when I found it on github was `http` so that's where it comes from.  Should we rename `REST`?  I'm not really partial....